docs/contributing/local-testing.md

```diff
@@ -0,0 +1,184 @@
+# Local testing of Claudie
+
+In order to speed up the development, Claudie can be run locally for initial testing purposes. However, it's important to note that running Claudie locally has limitations compared to running it in a Kubernetes cluster.
+
+## Limitations of Claudie when running locally
+
+### No Frontend/CRD testing
+
+The Frontend component heavily relies on the Kubernetes cluster and cannot be executed or tested locally. The same limitation applies to CRD testing since there is no environment to deploy them to.
+
+### No Autoscaling testing
+
+Testing or simulating the Claudie autoscaling is not feasible when running Claudie locally because it dynamically deploys Cluster Autoscaler and Autoscaler Adapter in the management cluster.
+
+### Claudie outputs
+
+Since Claudie generates two types of output per cluster (node metadata and kubeconfig), testing these outputs is not possible because they are created as Kubernetes Secrets.
+
+## Requirements to run Claudie locally
+
+As Claudie uses number of external tools to build and manage clusters, it is important these tools are installed on your local system.
+
+- `go` - check current version used in `go.mod` file
+- `terraform` - check current version used in Terraformer Dockerfile
+- `ansible` - check current version used in Ansibler Dockerfile
+- `kubeone` - check current version used in Kube-eleven Dockerfile
+- `kubectl` - check current version used in Kuber Dockerfile
+- `mongo` - when running locally, we recommend to run `mongo` as a container, check current version used in manifests for Mongo
+- `dynamo` - when running locally, we recommend to run `dynamo` as a container, check current version used in manifests for Dynamo
+- `minio` - when running locally, we recommend to run `minio` as a container, check current version used in manifests for Minio
+
+## How to run Claudie locally
+
+To simplify the deployment of Claudie into local system, we recommend to use rules defined in Makefile.
+
+To start all the datastores, simply run `make datastoreStart`, which will create containers for each required datastore with preconfigured port-forwarding.
+
+To start all services (excluding Frontend), run `make <service name>`, in separate shells. In case you will make some changes to the code, to apply them, please kill the process and start it again using `make <service name>`.
+
+## How to test Claudie locally
```

Comment:
Like mentioned above. You can freely deploy CRD and test the services with Kind local cluster. I think its worth mentioning.